grow/pods/collection_cache.py

```diff
@@ -0,0 +1,98 @@
+"""Cache for storing and retrieving data specific to collections.
+
+Supports caching the collection and caching documents within those collections.
+"""
+
+from . import collection
+import os
+import re
+
+class CollectionCache(object):
+
+    def __init__(self):
+        self.reset()
+
+    @staticmethod
+    def generate_cache_key(pod_path, locale):
+        return '{}::{}'.format(pod_path, locale)
+
+    def add_collection(self, col):
+        self._cache[col.collection_path] = {
+            'collection': col,
+            'docs': {},
+        }
+
+    def add_document(self, doc):
+        col = doc.collection
+        self.ensure_collection(col)
+        # NOTE: Using `doc.locale` causes infinite loop since it needs to load
+        # the fields (which can have circular dependencies).
+        cache_key = CollectionCache.generate_cache_key(
+            doc.pod_path, doc._locale_kwarg)
+        self._cache[col.collection_path]['docs'][cache_key] = doc
+
+        # Recache with the real locale once we have stored it.
+        new_cache_key = CollectionCache.generate_cache_key(
+            doc.pod_path, doc.locale)
+        if cache_key != new_cache_key:
+            self._cache[col.collection_path]['docs'][new_cache_key] = doc
+
+    def delete_by_path(self, path):
```

Comment:
This is referring just to the cache and not anything on the filesystem - right? If so, might be better to use "remove" rather than "delete". E.g. "add/remove".